packages/design-system-react/src/components/BadgeWrapper/BadgeWrapper.test.tsx

```diff
@@ -45,8 +49,11 @@ describe('BadgeWrapper', () => {
         <div data-testid="anchor" />
       </BadgeWrapper>,
     );
-    const badgeDiv = screen.getByTestId('badge').parentElement!;
-    expect(badgeDiv).toHaveStyle({ bottom: '0px', right: '0px' });
+    const badgeEl = screen.getByTestId('badge');
+    // Ensure the parent element exists
+    expect(badgeEl.parentElement).toBeInTheDocument();
+    // Apply styles check only after confirming parent exists
+    expect(badgeEl.parentElement).toHaveStyle({ bottom: '0px', right: '0px' });
```

Comment:
Addresses the `@typescript-eslint/no-non-null-assertion` violation by ensuring the parent element exists before accessing it, eliminating the need for a non-null assertion.